### PR TITLE
Various cleanups + document RM_OLD_IMAGE usage

### PR DESCRIPTION
**Pull request by mythi**
This PR adds various cleanups and documents RM_OLD_IMAGE usage.

@dbkinder please check a6abbe9
@brunobottazzini FYI: 577557a. Closes: #48

### CODE REVIEW

File: doc/howtos/building-images.rst [hunk 1]
```diff
@@ -229,3 +229,16 @@ build configuration.
 Alternatively, ``CORE_IMAGE_EXTRA_INSTALL`` can also be used. The
 difference is that this will also affect the initramfs images, which is
 often not intended.
+
+Automatic Cleaning of Old Image Before New One Get Built
```
Comment (dbkinder) [inline doc/howtos/building-images.rst]: How about: "Removing Previous Image to Save Disk Space"

File: doc/howtos/building-images.rst [hunk 2]
```diff
@@ -229,3 +229,16 @@ build configuration.
...
+--------------------------------------------------------
+
+Every image built gets copied in the deploy directory. Further, a new build
```
Comment (dbkinder) [inline doc/howtos/building-images.rst]: Every image build gets copied into the deploy directory. As you're developing, these repeated builds will start accumulating and use up more and more disk space. You can save disk space by removing previous images before the new one is built by adding this line to your local.conf:  RM_OLD_IMAGE = "1"

File: meta-ostro/recipes-soletta/soletta/soletta_%.bbappend [hunk 3]
```diff
@@ -1,6 +1,6 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
 
-DEPENDS_append = " systemd"
+DEPENDS_append = " systemd libmicrohttpd"
```
Comment (kad) [inline meta-ostro/recipes-soletta/soletta/soletta_%.bbappend]: shouldn't be going directly to meta-soletta ? this dependency management, why to have bbappend if we can put changes directly to meta-solleta ?

Comment (brunobottazzini) [inline meta-ostro/recipes-soletta/soletta/soletta_%.bbappend]: As far I know there is an open PR there: https://github.com/solettaproject/meta-soletta/pull/48.

We can merge that patch and remove the systemd depends here....

Comment (mythi) [inline meta-ostro/recipes-soletta/soletta/soletta_%.bbappend]: I just added the DEPENDS here to get a good build to try (to make sure the right version gets used).

I'm closely following meta-soletta#48

File: meta-ostro/conf/distro/ostro.conf [hunk 4]
```diff
@@ -206,6 +205,7 @@ PNWHITELIST_meta-java += " \
 PNWHITELIST_meta-soletta += " \
     soletta \
     soletta-dev-app \
+    libmicrohttpd \
```
Comment (brunobottazzini) [inline meta-ostro/conf/distro/ostro.conf]: @mythi, I have tested this patch to see if Soletta will build with http-server support.

Everything worked however, I had to use the patch solettaproject/meta-soletta#48 or yours here.

Thank you very much

Comment (mythi) [inline meta-ostro/conf/distro/ostro.conf]: On Thu, Mar 17, 2016 at 7:11 PM, Bruno Bottazzini <notifications@github.com>
wrote:

> In meta-ostro/conf/distro/ostro.conf
> <https://github.com/ostroproject/meta-ostro/pull/62#discussion_r56541190>:
>
> > @@ -206,6 +205,7 @@ PNWHITELIST_meta-java += " \
> >  PNWHITELIST_meta-soletta += " \
> >      soletta \
> >      soletta-dev-app \
> > +    libmicrohttpd \
>
> @mythi <https://github.com/mythi>, I have tested this patch to see if
> Soletta will build with http-server support.
>
> Everything worked however, I had to use the patch
> solettaproject/meta-soletta#48
> <https://github.com/solettaproject/meta-soletta/pull/48> or yours here.
>
> Thank you very much
>
>
Thanks, I'll drop 802e982.

**Comment by mythi** [comment]
rebased, dropped the DEPENDS change, and updated doc based on @dbkinder's feedback.

File: doc/howtos/building-images.rst (hunk 2 above)
Comment (pohly) [inline doc/howtos/building-images.rst]: On Thu, 2016-03-17 at 09:01 -0700, David Kinder wrote:
> by removing previous images before the new one is built by adding this
> line to your local.conf: RM_OLD_IMAGE = "1"

"after the new one is built" - if building somehow fails, the old one
remains and the symlink still points to it.

**Comment by kad** [comment]
retest this please

**Comment by mythi** [comment]
rebased, added 8aed707 and force pushed

**Comment by mythi** [comment]
@kad OK to merge (asking because it's my PR). Only reboot case fails on BBB.

**Comment by kad** [comment]
I think it's ok. :+1:  from my side.

**Comment by mythi** [comment]
OK, I'll merge.

**Pull request merged**